base_layer/core/src/covenants/output_set.rs

```diff
@@ -51,6 +51,7 @@ impl<'a> OutputSet<'a> {
         *self = new_set;
     }
 
+    /// Retains all outputs for which the given predicate f return true.
```

Comment:
Missing docs on `pub struct OutputSet` and public methods in `impl<'a> OutputSet<'a>` ?